### PR TITLE
Jetpack Settings: Introduce an updated custom content types card

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

This PR introduces an updated version of the Custom Content Types card. The old card was no longer matching the rest of the tab, since it was using the large toggles and its layout was quite different. This PR is part of #9171.

### What this PR does

* Introduces a new Custom Content Types card that works and looks just like the other cards in the same tab do. This is also outlined in p6TEKc-HQ-p2.
* Uses `wrapSettingsForm` and the rest of the already laid out infrastructure to implement the new card.
* Removes a ton of boilerplate code from the writing form, as well as the legacy card component.

### Preview

**The old Custom Content Types card**

**The new Custom Content Types card with enabled module**

**The new Custom Content Types card with disabled module**

### Notes

This PR includes #11095, which improves tests for the `filterSettingsByActiveModules()` methods.

### To test

* Checkout this branch
* Go to `/settings/writing/$site` where `$site` is a Jetpack site.
* Play around with the module toggle, enable and disable it and verify it works as expected.
* Play around with the Testimonials and Portfolio toggles and the Save Changes button and verify everything saves and loads correctly.
* Verify the card is not shown for Jetpack < 4.5
* Verify the card is not shown for WordPress.com sites.

/cc
@oskosk @roccotripaldi @johnHackworth @ryelle for code review
@MichaelArestad @rickybanister for design review

### CODE REVIEW

File: client/my-sites/site-settings/custom-content-types/index.jsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+						<ExternalLink href={ 'https://jetpack.com/support/custom-content-types' } target="_blank">
+							{ translate( 'Learn more about Custom Content Types' ) }
+						</ExternalLink>
+					</InfoPopover>
+				</div>
+
+				<JetpackModuleToggle
+					siteId={ siteId }
+					moduleSlug="custom-content-types"
+					label={ translate( 'Display different types of content on your site with custom content types.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/custom-content-types/index.jsx]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Use these settings to display different types of content on your site.' )` *ES Score: 5.13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
This feels so much better than the previous 'card within card'. That always frustrated me.

I think we can improve the italic descriptions, however. They are very dense. 

It might also be helpful to make the word 'Testimonial' and 'Portfolio' bold in the respective labels. It will stand out and make them easier to understand. 

I'll talk to @MichaelArestad about improving the description area.

**Comment by MichaelArestad** [comment]
@tyxla Let's remove the code tags around the shortcodes and we probably should do a second PR to darken the text. For some reason darkening the text makes it easier to read and feels less dense. Also, it would meet a11y guidelines. 🎉

**Comment by rickybanister** [comment]
It occurs to me that we could perhaps start to build out a universal 'preview' or 'example' component for settings. We could use that for the Related Posts, Ads, and a sort of Editor preview to show how these could be used. 

In a future PR of course.

**Comment by MichaelArestad** [comment]
@rickybanister Yep. I have the mockups somewhere. Creating the issue.

**Comment by tyxla** [comment]
@MichaelArestad @rickybanister I've done the updates you recommended, this one needs another design review.

> It might also be helpful to make the word 'Testimonial' and 'Portfolio' bold in the respective labels. It will stand out and make them easier to understand.

:+1: done in 0f0f775e784d5feae09c3290c6cf057b2b111a95, indeed looks much better.

> Let's remove the code tags around the shortcodes

:+1: done in 5f6776e14ea00541177f9c32b391c95ac3fd8a21.

> we probably should do a second PR to darken the text.

I've opened a PR for this: #11127.

**Comment by MichaelArestad** [comment]
🚢

**Pull request merged**